### DISCUSSION TITLE
NAT IP Private to IP Public

### DISCUSSION THREAD

**Discussion by dhodyrahmad** [strongswan/strongswan#321] [Q&A]
Hi, I'm working on GCP with centos 8. I need to make an access from specific IP that when hit to my domain (with HTTPS) and it will be arrive to my private IP on spesific port. Otherwise, when others hit my domain it still directed to my public IP and public directory.  

For example
```
If spesific IP:
192.168.x.x -> https://example.com -> IP_public -> NAT -> IP_private:port
```
```
If others IP:
Others IP -> https://example.com -> IP_public
```

Can I do this with strongswan or with NAT on strongswan?

**Comment by Thermi** [comment]
strongSwan is just an IKE keying daemon for IPsec, which is a layer three (IP layer) tunneling protocol. DNS names are on layer 7, so at the very least you will have to involve a DNS server or your local hosts file to change the IP address returned by the lookup for example.com.

**Reply by dhodyrahmad** [reply]
Hello @Thermi , Thank you for your response it helping me. What about NAT IP private to IP public? actually this is the main problem, some colleagues show me by using others VPN it can be NAT (from IP public it will translated to IP private) if there's traffic from specific IP (like a gave above). Any suggest with this problem, Sir?

So, the full stories are, I've done tunneling to third party and connection works well. The last step is, they want to access the tunnel using IP public with SSL. So, they suggest the NAT-ing from IP public to IP private like they've done (they using another VPN), But I had no idea how to do this on Strongswan.

**Reply by Thermi** [reply]
You need to use Netfilter/ipfw for that (depending on the OS you're running this on).

**Reply by dhodyrahmad** [reply]
I'm using centos 8 as vpn server. Netfilter is same as iptables on handling NAT, right?

**Reply by Thermi** [reply]
Netfilter is the implementation within the kernel.
iptables is a frontend to xtables, which is a framework for using Netfilter.

So yes, use iptables.
